### ISSUE TITLE
Column mode selection should not be part of multiple-selections

### ISSUE THREAD

**Issue by alankilborn** [notepad-plus-plus/notepad-plus-plus#13089]
### Description of the Issue

Multi-editing allows multiple stream selections to be made in succession, resulting in multiple selected text selections.  When a column-mode rectangular block selection is begun, any existing stream selections are unselected, and the column block is formed (correct behavior).  If a column block selection is in existence an an attempt is made to add a second column block to it, this also is not allowed--the first column block will be deselected first (correct again).

However, if a column block selection is in existence an an attempt is made to add one or more stream selections to it, to form a hybrid type of multiselection, this is allowed (incorrect behavior).

### Steps to Reproduce the Issue
1. Ensure _Settings_ menu > _Preferences..._ > _Editing_ > _Enable Multi-Editing_ is checkmarked.
2. Open Notepad++'s `license.txt` file.
3. Make the following column-block selection:
![image](https://user-images.githubusercontent.com/77065706/217950714-6051ba94-20a2-4f23-8b61-d7b207e807c7.png)
4. Hold down control while using the mouse to add this stream selection:
![image](https://user-images.githubusercontent.com/77065706/218087499-f325f667-d9c6-4768-83c9-8bb02da495cb.png)
5. Press Ctrl+c to copy the selected text.
6. Paste the text somewhere else.

### Expected Behavior

I should not have been allowed to add (in step 4) the stream selection to the existing column selection; the column selection should disappear (i.e., be cancelled) first, and then the stream selection started.
When a column block is used, no other additional selections should be allowed to be made simultaneously.

### Actual Behavior

As pictured in step 4 above.
After step 6 is executed, note that the text was pasted in column mode, i.e., as if the entire selection was a column mode block (disasterous).

### Debug Information
Notepad++ v8.4.9   (64-bit)
Build time : Jan 27 2023 - 03:11:16
Path : W:\npp.misc\_NoBackup\Releases\npp.8.4.9\npp.8.4.9.portable.x64\notepad++.exe
Command Line : -titleAdd="8.4.9 64-bit"
Admin mode : OFF
Local Conf mode : ON
Cloud Config : OFF
OS Name : Windows 10 Enterprise (64-bit) 
OS Version : 21H2
OS Build : 19044.2486
Current ANSI codepage : 1252
Plugins : 
    mimeTools (2.9)
    NppConverter (4.5)
    NppExport (0.4)
    PythonScript (2)

**Comment by Yaron10** [comment]
The inconsistency is interesting.
But isn't this a Scintilla issue?

**Comment by alankilborn** [comment]
>The inconsistency is interesting.
But isn't this a Scintilla issue?

I'm just logging the issue, first noticed using N++.  :-)
I haven't evaluated it in the Scintilla demo editor (if that is even possible).
A moderator can certainly tag it with a Scintilla-dependent label.

**Comment by Coises** [comment]
Are you sure about this?:
> After step 6 is executed, note that the text was pasted in column mode,

I still have 8.4.8 (32 bit) installed, and I do not see this. When I paste the copied selection, it pastes as a multiple stream selection, not as a rectangular selection; i.e., the selected text, without line endings added between selections, pastes at the location of the cursor (or cursors, if you have multiple selections, a rectangular selection or a thin selection when you paste).

To Scintilla, a rectangular selection is a special type of multiple selection. After step 4, the selection is converted from a rectangular selection to a multiple selection. I'm sure that's what's happening because I'm currently working on a plugin that tests, before doing certain operations, whether the selection is a rectangular selection. After step 4 the selection no longer passes that test (SCI_SELECTIONISRECTANGLE).

Notepad++ would have to test specifically to disallow this, and it's not clear to me that there's any reason it should. Selecting a rectangle first and then adding to it with additional multiple selections might occasionally be a useful shortcut for a user.

**Comment by alankilborn** [comment]
>>After step 6 is executed

>Are you sure about this

Well, I was at the time, but I see your point as I am unable to recreate it now.  So I retract that point.

----------------------

Maybe behavior at the paste is more unexpected by the casual user that it would be occasionally useful to a seasoned user?

If my steps to reproduce, in that order, are to be considered a valid and reasonable sequence, then shouldn't also making two stream selections followed by an (added) column block selection be equally valid?  Or making two or more column block selections in succession (so that both are then active)?

I think only one column block selection (standing alone) being allowed was the design intent, and that the behavior of this ticket is a bug.

But, opinions vary! :-)

----------------------

>I'm currently working on a plugin that tests, before doing certain operations, whether the selection is a rectangular selection. After step 4 the selection no longer passes that test (SCI_SELECTIONISRECTANGLE).

Unknown what your plugin does, but be aware if you're not already that a zero-width "rectangular" selection can test as SCI_SELECTIONISRECTANGLE true sometimes and false at other times.

**Comment by Coises** [comment]
I agree that it would make sense if either your sequence resulted in the step 4 stream selection replacing the rectangular selection _or else_ Ctrl+Alt+drag when there is an existing selection created a multiple selection (just as Ctrl+drag does, but adding from a rectangle rather than a stream). (It doesn't.)

Personally, I think that's a Scintilla anomaly which is just not significant enough that Notepad++ should try to compensate for it. It is reasonable to keep a note of it, though, for future reference.

Thanks for the heads-up about thin selections... I have come across them.
